tests/expressions_case/tests.py

```diff
@@ -1243,3 +1243,31 @@ def test_conditional_aggregation_example(self):
             ),
             {'regular': 2, 'gold': 1, 'platinum': 3}
         )
+
+    def test_annotation_with_values_and_when(self):
```

Comment:
Which tests are in the CaseDocumentationExamples ? Also the tests you attached are pretty much similar.I however attached two sets of tests one without and one with the annotated fields in the orderby to ensure better test coverage.